### PR TITLE
Update casing of "JabRef"

### PR DESCRIPTION
**Pull request by koppor**
Follow-up to https://github.com/JabRef/jabref/pull/10338.

We either need this fix - or add an explanation in the documentation how to "properly" check out the code to have IntelliJ working (one commit in https://github.com/JabRef/jabref/pull/10322 does this).

Since I more want to have the guideline "intuitive", I would like to have this change in.

The installer file names and the startup scripts will be all-lowercase now. Which I kind of like. (`bin/jabref` instead of `bin/JabRef`)

Update casing of "JabRef"

- Lowercase "jabref" in scripts and binary names
- Change vendor to JabRef e.V.
- Case "mac-package-identifier" to "JabRef"
- Group macos steps together

### Mandatory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by koppor** [comment]
Maybe, this conflicts with https://github.com/JabRef/jabref/pull/10340 (`JabRef.app` vs. `jabref.app` - to be checked)

File: .github/workflows/deployment-arm64.yml [hunk 1]
```diff
@@ -114,8 +114,8 @@ jobs:
           --verbose \
           --mac-sign \
           --vendor JabRef \
-          --mac-package-identifier Jabref \
-          --mac-package-name JabRef \
+          --mac-package-identifier JabRef \
+          --mac-package-name "JabRef e.V." \
```
Comment (Siedlerchr) [inline .github/workflows/deployment-arm64.yml:118]: Should go to vendor?

File: .github/workflows/deployment.yml [hunk 2]
```diff
@@ -283,26 +283,26 @@ jobs:
         if: steps.checksecrets.outputs.secretspresent == 'YES'
         id: gitversion
         uses: gittools/actions/gitversion/execute@v0.10.2
-      - name: Get macOS binaries
+      - name: Get macos binaries
```
Comment (Siedlerchr) [inline .github/workflows/deployment.yml:286]: This is wrong! The casing of macOS is correct official

**Comment by Siedlerchr** [comment]
We should discuss implications first

**Review by calixtus — DISMISSED** [review]

**Comment by Siedlerchr** [comment]
We should keep the casing as is, this how it looks like in mac: 
grafik

**Comment by koppor** [comment]
We should distinguish

- Linux: all lower case
- Mac OS: mixed case
- Windows: Don't know - mixed case?

#FutureWork

**Pull request closed without merging**